### PR TITLE
Jalmogo/dashboard

### PR DESCRIPTION
**Pull request by jalMogo**
This PR adds a dashboard:
 - adds some updates to our build steps
   - removes the `dist` dir, and flattens it down to the `www` dir
   - perform `rm -rf www/*` in the npm script, instead of across both the webpack and static-build scripts
 - update the session key in routes.init, instead of within the base.hbs template
 - adds a dashboard template, along with some charts as molecules

Sorry this PR is so long! It is deployed on snohomish and durham, which have dashboards configured. It is also deployed on argentina and palouse, which don't have dashboards configured (yet).

TODO:
 - [x] polish up layout issues
   - [x] re-align pie and bar charts
   - [x] relocate download data button?
 - [x] use proper labels for categories
 - [x] add percentages info to demographics bar chart
 - [x] make dashboard data driven
   - [x] do this just enough so that we can ship a (minimal) dashboard for other flavors, like Snohomish, Palouse, Argentina, etc
   - [x] update the "include private" checking logic to work when there are multiple datasets being fetched, eg: snohomish
   - [x] ~~or save it for a later pr, in which case we'll need to re-add the download data button to flavors that don't have a dashboard yet...~~
 - [x] test on other flavors for regressions
   - [x] snohomish staging
   - [x] durham staging
   - [x] argentina staging
 - [x] fix tests
 - [x] add a note about the demographics "multi-choice" data
 - [x] fix prod issue preventing dashboard from rendering

### CODE REVIEW

File: src/base/templates/base.hbs [hunk 1]
```diff
@@ -118,20 +119,14 @@
...
   </script>
 
   {{#if production}}
-    <script src="/dist/{{jsHashedBundleName}}"></script>
+    <script src="/{{jsHashedBundleName}}"></script>
   {{else}}
     <script src="/bundle.js"></script>
   {{/if}}
   <script src="{{config.app.api_root}}users/current?format=jsonp&callback=bootstrapCurrentUser"></script>
-  <script src="{{config.app.api_root}}utils/session-key?format=jsonp&callback=setApiSessionCookie"></script>
```
Comment (jalMogo) [inline src/base/templates/base.hbs]: setting the session key has been moved into the routes initialize step.

File: src/base/static/components/templates/dashboard.js [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+              dataKey="count"
+              stroke={BLUE}
+              activeDot={{ r: 8 }}
+            />
+          </LineChart>
+        </EngagementWrapper>
+        {this.props.dashboardConfig.surveyMetrics && (
+          <SurveyWrapper>
+            <RegularTitle style={{ gridArea: "title" }}>Survey</RegularTitle>
+            {/* TODO: Make these charts more data-driven. Right now, they are hard coded for durham */}
```
Comment (goldpbear) [inline src/base/static/components/templates/dashboard.js]: Was this TODO addressed?

Comment (jalMogo) [reply src/base/static/components/templates/dashboard.js]: No, but I punted it for now.

We will need to address it when we want to use the categories pie chart or the bar chart for other flavors.

Comment (goldpbear) [reply src/base/static/components/templates/dashboard.js]: Ah, ok-- sounds good.

**Comment by jalMogo** [comment]
Thanks for the feedback!

Bundle hashes and session id's should be fixed, and are on durham staging.

Feedback to your other suggestions are inline:

>I think the right side of the dashboard show have the same margin width as the left, similar to the layout of the list view. Right now the right side runs to the edge of the browser window (at least on my 1920x1080 screen).

I'm unable to repro this issue - I tried resizing the window, and using the 1920x1080 dimensions, and the content remains centered. Maybe you can show me this issue next time we chat.

>I think the title of the dashboard is a little too big-- nothing else in the app has that extra large font size. Maybe bring it down a notch?

done 👍

>I think the spacing below the header could be tighter with the horizontal rule below it

done 👍

>Do you think the dashboard should have a flavor-specific title? I feel like a generic name like "Analytics Dashboard" would be better, though I can't really put my finger on why

done 👍

>Perhaps the cards on the first row could be spaced tighter. They could be left-aligned, I think. Something about their layout now makes the dashboard feel sparse, though it just might be me.

I tweaked this using a "space-around" alignment instead of a "space-between". I'm open to more feedback, or we can always take another pass on the next iteration.

**Pull request merged**